python/lsst/ts/watcher/base_hexapod_overcurrent_rule.py

```diff
@@ -0,0 +1,190 @@
+                    type: number
+                    default: 4.0
+                max_count:
+                    description: >-
+                        Maximum count to check the hexapod current in idle.
+                        The telemetry rate is 20 Hz. Therefore, 10 mins are
+                        12000 times of telemetry.
+                    type: number
+                    default: 12000
+                severity:
+                    description: >-
+                        Alarm severity defined in enum AlarmSeverity.
+                    type: integer
+                    default: 2
+
+            required:
+            - threshold_current
+            - max_count
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    def compute_alarm_severity(
+        self, data: salobj.BaseMsgType, **kwargs: dict[str, typing.Any]
+    ) -> AlarmSeverityReasonType:
+        """Compute and set alarm severity and reason.
+
+        Parameters
+        ----------
+        data : `salobj.BaseMsgType`
+              Message from the topic described by topic_callback.
+        **kwargs : `dict` [`str`, `typing.Any`]
+            Keyword arguments. If triggered by `TopicCallback` calling
+            `update_alarm_severity`, the arguments will be as follows:
+
+            * topic_callback : `TopicCallback`
+              Topic callback wrapper.
+
+        Returns
+        -------
+        None, if no change or unknown, or a tuple of two values:
+
+        severity: `lsst.ts.xml.enums.Watcher.AlarmSeverity`
+            The new alarm severity.
+        reason : `str`
+            Detailed reason for the severity, e.g. a string describing
+            what value is out of range, and what the range is.
+            If ``severity`` is ``NONE`` then this value is ignored (but still
+            required) and the old reason is retained until the alarm is reset
+            to ``nominal`` state.
+
+        Notes
+        -----
+        You may return `NoneNoReason` if the alarm state is ``NONE``.
+        """
+
+        # Update the controller state and enabled substate
+        if hasattr(data, "controllerState"):
+            self._controller_state = ControllerState(data.controllerState)
+            self._enabled_state = EnabledSubstate(data.enabledSubstate)
+
+            if not self._is_enabled_and_stationary():
+                self._count = 0
+
+            return NoneNoReason
+
+        # Check the current and update the count
+        # In the normal operation, the motor current can be higher than the
+        # threshold. Therefore, we only care about the condition that we always
+        # have the motor current to be higher than the threshold all the time
+        # when there is no movement.
+        if self._is_enabled_and_stationary() and any(
+            [current >= self.config.threshold_current for current in data.motorCurrent]
+        ):
+            self._count += 1
```

Comment:
So, python won't limit the maximum size of an integer but this could really grow up indefinitely if unbounded like this. I suggest you add a maximum allowed value here. Maybe define a `MAX_COUNT` constant and then only add up if `self._count` is smaller than that?

something like:

```
MAX_COUNT = 2147483647  # This is actually the max value of int32.

...
            self._count += 1 if self._count < MAX_COUNT else 0
```

Reply:
Updated in the commit: [c789acd](https://github.com/lsst-ts/ts_watcher/pull/94/commits/c789acde31876242324909796fb547c8ff2cf2f0). Thanks!